### PR TITLE
Adds VB support for Unit Tests and WPF

### PR DESCRIPTION
**Pull request by KSchlobohm**
Addresses #384

### CODE REVIEW

File: tests/TestData/SmokeTests.LegacyMSTestVB/My Project/AssemblyInfo.vb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿Imports System
```
Comment (mjrousos) [inline tests/TestData/SmokeTests.LegacyMSTestVB/My Project/AssemblyInfo.vb:1]: Only the vbproj files should need committed, right? I think you can remove most of the test project and leave just the project files and still have the test work.

File: src/MSBuild.Abstractions/MSBuildConversionWorkspace.cs [hunk 2]
```diff
@@ -329,6 +329,13 @@ private bool IsSupportedProjectType(IProjectRootElement root, bool forceWeb)
                 Console.WriteLine($"{root.FullPath} contains an App.config file. App.config is replaced by appsettings.json in .NET Core. You will need to delete App.config and migrate to appsettings.json if it's applicable to your project.");
             }
 
+            if (!keepCurrentTFMs
+                && root.PropertyGroups.Any(pg => pg.Properties.Any(ProjectPropertyHelpers.IsVisualBasicProject))
+                && root.ItemGroups.Any(ig => ig.Items.Any(ProjectItemHelpers.IsReferencingSettingsSingleFileGenerator)))
+            {
+                Console.WriteLine($"{root.FullPath} uses code generators which will not be handled by try-convert. You can edit your vbproj to add support or remove these dependencies.");
+            }
```
Comment (mjrousos) [inline src/MSBuild.Abstractions/MSBuildConversionWorkspace.cs:337]: You don't return here. Just to confirm, you intend to give the warning message in this case but still allow try-convert to execute?

File: src/MSBuild.Abstractions/ProjectItemHelpers.cs [hunk 3]
```diff
@@ -124,5 +124,12 @@ public static class ProjectItemHelpers
         public static bool IsReferencingSystemWeb(ProjectItemElement item) =>
             item.ElementName.Equals(MSBuildFacts.MSBuildReferenceName, StringComparison.OrdinalIgnoreCase)
             && item.Include.Equals(MSBuildFacts.SystemWebReferenceName, StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsReferencingSettingsSingleFileGenerator(ProjectItemElement item)
+        {
+            var metadata = item.Children.FirstOrDefault(child => "Generator".Equals(child.ElementName, StringComparison.Ordinal) ) as ProjectMetadataElement;
```
Comment (mjrousos) [inline src/MSBuild.Abstractions/ProjectItemHelpers.cs:130]: These strings should probably be pulled out into constants in `MSBuildFacts` or some similar facts class that's appropriate.

File: src/MSBuild.Conversion.Facts/MSBuildFacts.cs [hunk 4]
```diff
@@ -246,10 +246,12 @@ public static class MSBuildFacts
             "ProjectReference"
         );
 
+        public static Guid LanguageProjectTypeVisualBasic => Guid.Parse("{F184B08F-C81C-45F6-A57F-5ABD9991F28F}");
```
Comment (mjrousos) [inline src/MSBuild.Conversion.Facts/MSBuildFacts.cs:249]: This is the guid that was labeled F# previously. What that comment wrong or do we need to swap this guid?

**Review by mjrousos — COMMENTED** [review]
Looks good; I left some comments inline.

File: src/MSBuild.Abstractions/MSBuildConversionWorkspace.cs (hunk 2 above)
Comment (KSchlobohm) [reply src/MSBuild.Abstractions/MSBuildConversionWorkspace.cs:337]: Correct, the intention is to allow the user to proceed with a warning that we expect a compile error after the conversion. I believe this is similar in experience to how we handle web projects.

File: src/MSBuild.Conversion.Facts/MSBuildFacts.cs (hunk 4 above)
Comment (KSchlobohm) [reply src/MSBuild.Conversion.Facts/MSBuildFacts.cs:249]: I believe the comment was wrong.

File: tests/TestData/SmokeTests.LegacyMSTestVB/My Project/AssemblyInfo.vb (hunk 1 above)
Comment (KSchlobohm) [reply tests/TestData/SmokeTests.LegacyMSTestVB/My Project/AssemblyInfo.vb:1]: That feedback makes sense. I decided to follow the established pattern which seemed to include the full project.

I'm referencing these projects:
* SmokeTests.MStestCoreBaseline
* SmokeTests.WpfNet5Baseline

**Comment by KSchlobohm** [comment]
@cartermp for review

File: src/MSBuild.Conversion.Facts/MSBuildFacts.cs (hunk 4 above)
Comment (cartermp) [reply src/MSBuild.Conversion.Facts/MSBuildFacts.cs:249]: if I recall, this was pulled from a legacy F# project. Interesting!

**Review by cartermp — APPROVED** [review]
Thanks! I think this looks good. I will merge this in now and I'll publish a release next week.

**Pull request merged**